tower-http/src/set_response_header.rs

```diff
@@ -0,0 +1,454 @@
+//! Set a header on the response.
+//!
+//! The header value to be set may be provided as a fixed value when the
+//! middleware is constructed, or determined dynamically based on the response
+//! by a closure. See the [`MakeHeaderValue`] trait for details.
+//!
+//! # Example
+//!
+//! Setting a header from a fixed value provided when the middleware is constructed:
+//!
+//! ```
+//! use http::{Request, Response, header::{self, HeaderValue}};
+//! use tower::{Service, ServiceExt, ServiceBuilder};
+//! use tower_http::set_response_header::SetResponseHeaderLayer;
+//! use hyper::Body;
+//!
+//! # #[tokio::main]
+//! # async fn main() -> Result<(), Box<dyn std::error::Error>> {
+//! # let render_html = tower::service_fn(|request: Request<Body>| async move {
+//! #     Ok::<_, std::convert::Infallible>(Response::new(request.into_body()))
+//! # });
+//! #
+//! let mut svc = ServiceBuilder::new()
+//!     .layer(
+//!         // Layer that sets `Content-Type: text/html` on responses.
+//!         //
+//!         // We have to add `::<_, Body>` since Rust cannot infer the body type when
+//!         // we don't use a closure to produce the header value.
+//!         //
+//!         // `if_not_present` will only insert the header if it is not present already.
```

Comment:
nit: maybe
```suggestion
//!         // `if_not_present` will only insert the header if does not already have a value.
```
or something? just to make sure it's clear that this won't insert the header if _any_ value for that header is already present, not the specific value.